### PR TITLE
Added httpConnectionManagerConfigHelper to refactor config_test

### LINKED ISSUE #11109: test: refactoring opportunities for FakeUpstream and health_checker_impl_test.cc
In the process of https://github.com/envoyproxy/envoy/pull/11098 I found there were large amounts of repeated code constructing FakeUpstream and in the health-checker test code.

IMO this boilerplate should be factored out into helper methods, greatly reducing code duplication and making future refactors much easier. For example, in test/common/upstream/health_checker_impl_test.cc there are tons of blocks exactly like

```
    health_checker_ = std::make_shared<TestHttpHealthCheckerImpl>(
        *cluster_, parseHealthCheckFromV2Yaml(yaml), dispatcher_, runtime_, random_,
        HealthCheckEventLoggerPtr(event_logger_), http_context_);
```

that could be factored into a single line, reducing code bloat, increasing (IMO) reliability, and certainly making future refactors easier. E.g. see the repetitive changes in https://github.com/envoyproxy/envoy/pull/11098/files#diff-441f309ff25aa293fedfc913fbd86fda

Similarly, there a lot of places across the codebase where a FakeUpstream is instantiated, and most (though not all) of the calls are quite similar, using already-set member variables for most of the arguments. There would be some parameterization needed. I count 33 instances to `new FakeUpstream(` which could also be factored.

A third case is test/extensions/filters/network/http_connection_manager/config_test.cc with 
```
HttpConnectionManagerConfig(parseHttpConnectionManagerFromV2Yaml(yaml_string), context_,
                                  date_provider_, route_config_provider_manager_,	                                  date_provider_, route_config_provider_manager_,
                                  scoped_routes_config_provider_manager_, http_tracer_manager_),	                                  scoped_routes_config_provider_manager_, http_tracer_manager_,
                                  http_context_)
```

This might be a good intern starter project.

### PR DESCRIPTION
**Pull request by mk46**
Signed-off-by: Manish Kumar <manish.kumar1@india.nec.com>

Commit Message: Refactored config_test.cc  
Additional Description:
created createHttpConnectionManagerConfig() to improve readability.

Risk Level:
Testing: unit test
Docs Changes:
Release Notes:
 includes partial fix to #11109

### CODE REVIEW

**Comment by dio** [comment]
@mk46, thanks! Could you update the PR description? Thanks!

File: test/extensions/filters/network/http_connection_manager/config_test.cc [hunk 1]
```diff
@@ -56,6 +56,12 @@ class HttpConnectionManagerConfigTest : public testing::Test {
   NiceMock<Tracing::MockHttpTracerManager> http_tracer_manager_;
   std::shared_ptr<NiceMock<Tracing::MockHttpTracer>> http_tracer_{
       std::make_shared<NiceMock<Tracing::MockHttpTracer>>()};
+  void httpConnectionManagerConfigHelper(
```
Comment (dio) [inline test/extensions/filters/network/http_connection_manager/config_test.cc:59]: I think this should be something like `createHttpConnectionManagerConfig`. WDYT?

Comment (mk46) [reply test/extensions/filters/network/http_connection_manager/config_test.cc:59]: @dio Thanks for pointing it. May I force push now or wait for the ci test?

**Comment by mk46** [comment]
Hi @dio, I found 
`HttpConnectionManagerConfig config(parseHttpConnectionManagerFromV2Yaml(yaml_string), context_,
                                     date_provider_, route_config_provider_manager_,
                                     scoped_routes_config_provider_manager_, http_tracer_manager_);`
is 48 times. Could you suggest how to refact it? Thanks!

File: test/extensions/filters/network/http_connection_manager/config_test.cc (hunk 1 above)
Comment (dio) [inline test/extensions/filters/network/http_connection_manager/config_test.cc:59]: Can we have createHttpConnectionConfig with a yaml string as an input?

Comment (mk46) [reply test/extensions/filters/network/http_connection_manager/config_test.cc:59]: @dio I will do that but what about  
`HttpConnectionManagerConfig config(parseHttpConnectionManagerFromV2Yaml(yaml_string), context_,
                                     date_provider_, route_config_provider_manager_,
                                     scoped_routes_config_provider_manager_, http_tracer_manager_);`
?

**Comment by dio** [comment]
@mk46 sorry, the Commit Message in the description, please? As per: https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md#commit-message

**Comment by mk46** [comment]
@dio PTAL

**Comment by dio** [comment]
Sorry, @mk46 could you merge master, please? Thanks!

**Pull request merged**